Feature Committed podcast banner and page

kimberleyworld opened the pull request:
Added
- Committed banner for home page 
- Committed page

Changed 
- Podcast_series_pre_get_posts() to look at serial categories array

kimberleyworld committed:
Formatting options-front-page.php

kimberleyworld committed:
format podcast-foreign-agent.php

kimberleyworld committed:
Creating files

Duplicated the content from the Podcast: Foreign Agent structure and changed the name to check I can see it on staging site

kimberleyworld committed:
Test edit

kimberleyworld committed:
Dummy info looking at how to edit

Ready for designs to update banner

kimberleyworld committed:
Removed svg

kimberleyworld committed:
Formatted catagory-forign-agent.php

Formatted and then duplicated in new file catagory-committed.php

kimberleyworld committed:
Checking I can update the new post catagory

kimberleyworld committed:
Initial work on the banner

- wrong image in
- not aligned properly

kimberleyworld committed:
Adjusting banner to match design

kimberleyworld committed:
Adjusted font sizes and zoomed into background img

kimberleyworld committed:
Staging site check

Adjusting zoom

kimberleyworld committed:
Design review edits

Review with design team

kimberleyworld committed:
Added responsivity to committed page

kimberleyworld committed:
Update functions-hooks.php

- Formatted file
- Adjusted  podcast_series_pre_get_posts function to accept an array and added committed

kimberleyworld committed:
Formatted functions.php

kimberleyworld committed:
Removed redundant classes

kimberleyworld committed:
Implementing design changes commited page

Started building credits section, built listen now section.

kimberleyworld committed:
made credits responsive

kimberleyworld committed:
adjustments from feedback

kimberleyworld committed:
Made credits box full width

kimberleyworld committed:
ensuring code is efficient

kimberleyworld committed:
reduced img sizes

kimberleyworld committed:
Code review and optimise banner

kimberleyworld committed:
Adjusted reporters image size

kimberleyworld committed:
Update CHANGELOG.md

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 5980dbf according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/332

kimberleyworld committed:
Fixing code scanning problems

- incomplete string escaping line 27372

kimberleyworld committed:
Merge branch 'feature-commited-podcast-banner-and-page' of https://github.com/novaramedia/novaramedia-com into feature-commited-podcast-banner-and-page

varffs commented on CHANGELOG.md:
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
+
+### Added
+
+- Committed banner for home page
```
This can just be one line but worth clarifiying what `Committed` means. Something like 

- Committed (podcast series) archive page and banner

varffs commented on CHANGELOG.md:
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
+
+## Changed
+
+- Podcast_series_pre_get_posts() to look at serial categories array
```
Don't need to capitalise this as the function name isn't capitalised makes sense to match it exactly

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,313 @@
+<main id="main-content" class="category-archive">
+  <style type="text/css">
+    .webp .committed__backgrounded {
+      background-image: url(<?php echo $base_image_path . 'catagory-committed-background.webp'; ?>);
```
maybe doesnt matter but typo in `catagory`/`category`

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,313 @@
+      .committed-credits__logo {
+      height: 45px;
+    }
+    .committed-credits__reporters img{
```
misaligned indentation

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,313 @@
+    </div>
+    <div class="flex-grid-row mt-4 mb-5">
+      <div class="flex-grid-item flex-offset-s-0 flex-item-s-12 flex-offset-l-1 flex-item-l-10 flex-offset-xxl-2 flex-item-xxl-8 font-serif committed__serif-large text-paragraph-breaks">
+        <p>In the 1970s, the Provisional IRA was in the early days of its armed campaign to end British rule on the island of Ireland. In the United States, a small group of activists began organising on their behalf. They called themselves the Irish Northern Aid Committee, or Noraid—and they were looking for a fight.</p>
```
This being the copy for Foreign Agent suggests not yet ready to merge? I think best to complete everything (or make this copy user editable via the CMS) before merging

kimberleyworld replied:
understand as we will need to go live with this at a later date I wanted you to have looked at the code. I have a list of all the things to edit before it goes to the actual site but we don't have the real copy or links yet

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,313 @@
+        <div class="committed-archive__box">
+          <div class="committed-listen-now-title mb-1"></div>
+          <div class="font-weight-bold committed-listen-now__links font-size-13 font-size-s-12">
+            <a href="https://podcasts.apple.com/us/podcast/committed-the-iras-american-connection/id1624937065?uo=4">Apple Podcasts</a>,
```
Obv these links will need to be updated before going live. I would think sensible to remove the def incorrect Foreign Agent ones just incase they accidentally get set live.

kimberleyworld replied:
I have a list of things to change, we don't have the links yet but yes will do

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,313 @@
+        <!-- Producers -->
+        <div class="flex-grid-item flex-item-m-4 flex-item-l-3 flex-item-xxl-3 flex-item-s-8 mb-s-5">
+          <div class="font-weight-bold mb-1">Producers</div>
+            <p class="mb-2"><?php echo nl2br( htmlspecialchars( $credits['producers'] ) ); ?></p>
```
Not sure you need the `nl2br` & `htmlspecialchars` combo here. If you do need formatting likely using the apply_filters the_content filter would be best. That is a core Wordpress way to do something similar. If you search for that in the codebase you should find some examples

kimberleyworld replied:
this pattern made sense to me in the moment, but it's not user entered data so your method makes way more sense

varffs commented on category-foreign-agent.php:
```diff
@@ -2,43 +2,51 @@
+ *
+ * @return void
+ */
+function nm_render_podcast_credit( $credit ) {
```
I realise that this function name is recycled (or at least the same on both of these archive pages). Perhaps that doesnt matter at all, or perhaps both instances should have unique function names? Maybe worth some quick research. Likely it just gets set like this in each file and is overridden to the intended version whenever needed, but maybe it is bad practice?

kimberleyworld replied:
its only loaded when that file is run but after research it says that plugin systems might assume function names are unique so I've changed it just in case.

varffs commented on category-foreign-agent.php:
```diff
@@ -114,7 +122,7 @@ function nm_render_podcast_credit($credit) {
     <div class="flex-grid-row mb-4">
       <div class="flex-grid-item flex-item-xxl-12 mb-4">
-        <h4 class="font-size-9 text-uppercase font-weight-bold" style="color: #E3FFFF;"><a href="<?php echo home_url('category/audio/'); ?>">Podcast</a></h4>
+        <a href="<?php echo home_url( 'category/audio/' ); ?>" class="ui-tag-block"><span class="ui-tag">Podcast</span></a>
```
Because of its usage here on top of imagery probably should be the varient used on primary featured images with no border and the other hover state. But this is a design detail so perhaps intentional/to be checked by Pietro.

varffs replied:
Same on the new Committed page

kimberleyworld replied:
Its been through design review and passed but i will flag this specific bit

varffs commented on dist/img/specials/committed/committed-background.jpg:
Is this file used in the end?

kimberleyworld replied:
yes, as there is no transparency i opted for jpg

varffs commented on dist/img/specials/committed/catagory-committed-background.png:
The build scripts are a bit clunky but if you put all files in the `src` path image folders first and then run the build script you get automatically optimised versions in the `dist` paths. Worth doing normally as I think by eye some of these files could be smaller. But also sometimes that build script isn't optimal so fine to override it if needed. Certainly 1.5mb file here is bigger than idea. Not sure about the avif and webp compression here tho as don't think the build scripts do those file types yet

varffs commented on functions.php:
```diff
@@ -1,19 +1,21 @@
-  	'liveCheckerData' => nm_get_livechecker_data()
+      'siteUrl'                  => home_url(),
+      'themeUrl'                 => get_template_directory_uri(),
+      'isAdmin'                  => current_user_can( 'manage_options' ) ? 1 : 0,
```
why the change to `manage_options`?

kimberleyworld replied:
there was some errors when I tried to create the pull request and this was one of the things it wasn't happy with: capability-based checks like manage_options are more WordPress-standard and future-proof.

varffs commented on lib/functions-hooks.php:
```diff
@@ -8,96 +8,96 @@
-function podcast_series_pre_get_posts($query) {
-  if ($query->is_admin()) { // ignore admin queries
+function podcast_series_pre_get_posts( $query ) {
+  if ( is_admin() ) {
```
why are you removing the `$query->` targeting here for these functions?

kimberleyworld replied:
Another thing I refactored due to errors becasue  is_admin() is a global function, I thought it doesn't need the $query object to work. It checks the context of the current request — not the specific query object.

varffs commented on lib/functions-hooks.php:
```diff
@@ -8,96 +8,96 @@
-      return;
+  $serial_categories = array( 'foreign-agent', 'committed' ); // Add more slugs as needed
+
+  if ( $query->is_archive() && is_category( $serial_categories ) ) {
```
same question as above with the `is_category`

kimberleyworld replied:
You are right, my logic above is correct but this one isn't,` if ( is_admin() ) { return; }` is global context but the query check with` $query->is_category() and $query->is_archive()` is specific to the $query object

varffs commented on lib/theme-options/options-front-page.php:
```diff
@@ -51,522 +54,638 @@ function nm_register_front_page_options_metabox() {
+      'partials/specials/banners/focus-breaking-britain' => 'Focus: Breaking Britain',
+      'partials/specials/banners/focus-disability-its-political' => 'Focus: Disability: It’s Political',
+      'partials/specials/banners/podcast-planet-b'       => 'Podcast: Planet B',
+      'partials/specials/banners/podcast-committed'      => 'Podcast: Committed',
```
Tend to add these in reverse chrono order but leaving the support stuff at the top. So better to put this above If I Speak

varffs commented on partials/specials/banners/podcast-committed.php:
```diff
@@ -0,0 +1,96 @@
+      <div class="grid-item is-s-12 is-xxl-6 is-m-7 committed-banner__grid-item committed-banner__grid-item-right mt-s-4">
+        <div class="committed-banner__button-container committed-banner__grid-item-right">
+          <p class="font-color-white font-weight-bold font-size-11 mb-3 pt-6 pt-s-0 font-size-s-11">Meet the people who</br>have done just that.</p>
+          <a href="<?php echo esc_url( $url ); ?>" class="ui-button ui-button--small ui-button--white">Listen Now</a>
```
Design/UX point but probably all the main elements of the banner should be linked not just the button

varffs reviewed (changes requested):
A few details here and there. But from looking at this on my dev machine and seeing the dummy content this is not really ready to merge as not ready yet. Fine and good to keep it as a branch in development until completely finished with content and everything. Also assuming that it hasn't had a pass for design detailing yet? Will be worth using the staging site and pulling this branch to it and getting a usable version that Pietro and co and look at for design details.

I also note that the permalink redirecting isn't set up yet as per Foreign Agent. I can't remember where or how that works but if you check the CMS and try to go to a permalink of a single episode of FA you will notice the redirect to the single full page archive page. If Committed follows that same design convention it will need that same navigation UX.

kimberleyworld commented:
> Member
I should have been more clear, I have been following the process and this has been through all the necessary design reviews on the staging site. I wanted you to look at the code so that when the actual podcast is ready to go live I can update the links and go with it straight away. I wasn't intending to merge, perhaps i should have just asked you to look at the branch

The single page view of a podcast hasn't been discussed and we don't have a design so we may have over looked that, I will check with the others. edit - i think i understand what you mean now, i'll investigate how it works

kimberleyworld committed:
Edits from patrick PR

Main bulk of the pr edits, a couple more to do

kimberleyworld committed:
Corrections pr

- formatted single page
- created a redirect function for single post to archive
- created links from banner to committed page
- links to contributor page
- fixed CodeQL js/incomplete-sanitization error main.js

kimberleyworld committed:
removed redundant styles and created a redirect

- Tidied up some of the old styles I noticed.
- fixed a breaking linting problem in the classesToSelector function
-Added  a redirect to /committed
- removed old breakpoint variables

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,317 @@
+    'thanks'         => 'All of the prisoners, their partners, friends and family members who spoke to us for this podcast. Thanks to their prison buddies: Alex, Jamie, Kate and Pia, and to Bertie Coyle from Just Stop Oil.',
+);
+
+$base_image_path = get_stylesheet_directory_uri() . '/src/img/specials/committed/';
```
Sorry to be clear I meant put the working files in src (meaning source) and then run the `build` script on the repo and it will generate final optimised files and put them in dist (meaning distribution) and then you reference those files in the actual output. So yes the repo does end up with duplicated files but the src folder is kinda disposable and unimportant and all the production stuff comes from dist (like the way that css and js works here). The difference is that to avoid wasting resources the `dev` script doesnt watch and optimise images all the time.

kimberleyworld replied:
I see, ok, i misunderstood

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,317 @@
+  <section class="container">
+    <div class="flex-grid-row mb-4">
+      <div class="flex-grid-item flex-item-xxl-12 flex-item-m-10 mb-4 mt-4">
+        <a href="<?php echo home_url( 'category/audio/' ); ?>" class="ui-tag-block ui-tag-block--no-border"><span class="ui-tag">Podcast</span></a>
```
I had a further thought about this. The tags in most contexts use the `render_post_ui_tags` function which obv cant work here. However the point of it being a function not markup is so that we can make minor changes easily. I get that we might want to do quite a lot of this kind of usage where it is really 3 variables of label, link and variant classes. I think it would be good to make a utility renderer function for this (and Foreign Agent) page(s). 

I'm not sure if it will make sense to then refactor the existing `render_post_ui_tags` function to use the new very simple outputting function needed here as the actual code that outputs. Good to keep the more complex querying and other logics in that separate function rather than just having 1 mega function. 

In this way this markup would only need to be generated by 1 function and thus changing that markup would be very easy.

kimberleyworld replied:
I've made a render_specials_ui_tag() function in renderers.php and implements in foreign agent and committed

kimberleyworld replied:
changed to render_series_ui_tag() as made more sense

varffs commented on src/styl/layouts/above-the-fold.styl:
```diff
@@ -18,15 +18,15 @@
   grid-row-end: span 1;
 
-// @media screen and (max-width: $breakpoint-m)
+// @media screen and (max-width: 910px)
```
revert these to stylus variables as per PR on the library

varffs commented on lib/functions-custom.php:
```diff
@@ -1,4 +1,50 @@
 <?php
+/**
```
Please put the rewriting type functions in the `functions-rewrites.php` file so all that type of stuff happens in the same place

varffs commented on lib/functions-custom.php:
```diff
@@ -1,4 +1,50 @@
+ *
+ * @return void Exits script execution after issuing a redirect.
+ */
+function nm_serial_podcast_redirect() {
```
This is good as refactor, but it should be using `add_rewrite_rule` rather than `wp_redirect`. This is because the redirect actually does 2 page loads and jumps from one to the other. Fine for temporary stuff, form submission or going to a different domain completely. For this if we use `add_rewrite_rule` we should get smooth routing from the beginning. Wordpress triggers against that configuration earlier in the processing of each page request.

varffs commented on lib/functions-custom.php:
```diff
@@ -1,4 +1,50 @@
+ * @return void
+ */
+function redirect_committed_custom_url() {
+  if ( isset( $_SERVER['REQUEST_URI'] ) ) {
```
You can basically copy exactly the config in the `functions-rewrites.php` file that does this for novara live. (Maybe suggests we should refactor that code a bit to just do this from an array or something rather than repeating code)

kimberleyworld replied:
I've added a comment to refactor into an array

kimberleyworld committed:
new resized reporter img and credit spacing

kimberleyworld committed:
Revert "removed redundant styles and created a redirect"

This reverts commit d90072aaf6526cecdb0339b7fc1319e7f825b98f.

kimberleyworld committed:
Added hover state to all banner items

kimberleyworld committed:
Added redirect function back

varffs commented on src/img/specials/committed/committed-clare.jpg:
While the src>dist pipeline does compression etc it doesnt scale images. So you will want to scale these down to an appropriate size for use on the site. Probably 2x the maximum displayed size in that layout

kimberleyworld replied:
ah yes, these were a last min swop yesterday and i forgot to check the file size

kimberleyworld committed:
Updated to nm-stylus-library#v0.10.0

kimberleyworld committed:
PR edits

- moved redirect to rewrites and refactored
- created a specials ui-tag renderer function
- changed image path back to src
- reduced size of reporter imgs

kimberleyworld committed:
ran dev build

kimberleyworld committed:
main description committed

kimberleyworld committed:
Added image credits

kimberleyworld committed:
estimated listen now links

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,330 @@
+          <?php echo wp_kses_post( $credits['thanks'] ); ?>
+
+          <div class="font-weight-bold mt-5 mb-1">Image credits</div>
+          <?php echo wp_kses_post( $credits['image_covers'] ); ?>
```
No spaces or line breaks for these credits? Probably the headings here should be h5s or h6s as that is more semantic, but not sure exactly how that would stack down the flow of the page

kimberleyworld replied:
Added line breaks and swopped to h6

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,330 @@
+          <div class="committed-listen-now-title mb-1"></div>
+          <div class="font-weight-bold committed-listen-now__links font-size-13 font-size-s-12">
+            <a href="https://podcasts.apple.com/us/channel/novara-media/id6742787656">Apple Podcasts</a>,
+            <a href="https://open.spotify.com/show/3KtmyPhvQ2FvhwdRNo5I1N?si=a95692ce1d75432d">Spotify</a>,<br/><a href="#">Google Podcasts</a>,
```
Realised Google Podcasts doesnt even exist anymore. These urls are unlikely to be the final ones because I think it will be a separate feed, but only 4 episodes so maybe not

kimberleyworld replied:
yep we were in coms about the links but chal has given me the correct ones now. google podcast doesn't exist you're right, rip

kimberleyworld replied:
we are using the main nm feed for now and then going to do a patch release to update to a separate feed or the first episode in a few weeks.

varffs commented on category-committed.php:
```diff
@@ -0,0 +1,330 @@
+      <article class="flex-grid-row pt-6 pt-s-4" id="<?php echo $post->post_name; ?>">
+        <div class="flex-grid-item flex-offset-s-0 flex-item-s-12 flex-offset-xxl-1 flex-item-xxl-4 mobile-mb-4">
+          <h4 class="font-size-9 text-uppercase font-weight-bold mb-2 mb-s-0"><?php echo $meta['_cmb_standfirst'][0]; ?></h4>
+          <h3 class="font-size-13 font-weight-semibold js-fix-widows"><?php the_title(); ?></h3>
```
I've mostly given up on fix widows function now due to a few issues with it so can be removed.

varffs commented on dist/img/specials/committed/catagory-committed-background.png:
1.33mb for this quite small file seems really high. As long as the grain is kept so it will scale well I think it could be much more compressed

varffs commented on dist/img/specials/committed/committed-background.jpg:
I cant see where this image is used. If it is used it is very big both in dimensions and also filesize

varffs replied:
Sorry I see its on the banner. At 3.19mb its way too big to be loading on the homepage. The homepage especially needs to be tight with resource use as to make sure we get good speeds for google etc. So on the homepage there quality to speed balance should be even more speed. This wants to be 250kb or less ideally, maybe up to 500 but thats a bit much I think. Not sure other banner resource image sizes so maybe worth checking. Easiest fix is dimensions as this is a large image, but also compression.

kimberleyworld replied:
I hadn't noticed how big it was, i re-exported and compressed, do you find it easier to export things your self so you can make them the right format or would you normally ask for an assets file from Pietro (and now fillipo)

varffs replied:
Tbh I think I find it easier to do resource export and manipulation myself, but there isn't really any reason it should land on the developer in the flow. I think ideally assets should be at correct scale in Figma and therefore it should be possible just to export at 2x from Figma designs and drop into the `src` folder and work from there. I think it is reasonable from your position to request correctly scaled and perhaps even compressed images (tho compression less of a deal due to the build tool in the repo)

varffs commented on dist/img/specials/committed/committed-clare.jpg:
This is unscaled still so needs to get smaller dimensions which should help drop the filesize too

kimberleyworld replied:
yes it's sorry, again this was a last min swop in. Will keep file size and actual size top of my mind from now on

varffs commented on lib/renderers.php:
```diff
@@ -1,5 +1,18 @@
+ * @param string $url The link to wrap the tag in.
+ * @param string $variant_class Optional extra classes can be added for styling variants.
+ */
+function render_series_ui_tag( $label, $url, $variant_class = '' ) {
```
There isnt really anything about this that is specific to being a series tag, so the function could be named more general for potential use in other locations

varffs commented on partials/specials/banners/podcast-committed.php:
```diff
@@ -0,0 +1,109 @@
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      padding: 0px;
```
The padding probably shouldnt be removed from grid items as it will create vertical misalignments on the outside edges with other grid layouts on the page. It also means in the `m` scale of this banner the first and second grid elements directly touch on my screen which looks uncomfortable

varffs reviewed (changes requested):
Mostly need to get the image sizes down, mostly by scaling dimensions. But also still need final content to merge with for the soundcloud trailer and podcast follow links.

kimberleyworld committed:
Finale comments from pr

- resized and compressed background anrd reporter images
- refactored function to render_ui_tags
-  removed padding 0 in grid
- Checked links for listen now

kimberleyworld committed:
Merge branch 'development' into feature-commited-podcast-banner-and-page

varffs commented on lib/renderers.php:
```diff
@@ -6,7 +6,7 @@
-function render_series_ui_tag( $label, $url, $variant_class = '' ) {
+function render_ui_tag( $label, $url, $variant_class = '' ) {
   ?>
   <a href="<?php echo esc_url( $url ); ?>" class="ui-tag-block ui-tag-block--no-border <?php echo esc_attr( $variant_class ); ?>">
```
This class `ui-tag-block--no-border` is already the `$variant_class` essentially so really should be being passed as a function parameter. Feels like it would be nice to have the function actually just take `no-border` as the function parameter and then render the class out in the function, but probably then the param would need to be an array of variant classes which would all be rendered out. I don't actually know if there are other variant classes for the ui tag at the moment anyway

varffs reviewed (commented):
Just noticed something on the ui tag function but otherwise everything is good

kimberleyworld committed:
Correct trailer link

- $variant_class being passed as a function parameter
- centred listen now box
- updated change.log

kimberleyworld committed:
Merge branch 'feature-commited-podcast-banner-and-page' of https://github.com/novaramedia/novaramedia-com into feature-commited-podcast-banner-and-page

kimberleyworld committed:
Added official description
